mdx-models/src/main/java/com/mx/path/model/mdx/model/account/ExtendedFields.java

```diff
@@ -0,0 +1,21 @@
+package com.mx.path.model.mdx.model.account;
+
+import java.util.List;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import com.google.gson.annotations.SerializedName;
+import com.mx.path.model.mdx.model.MdxBase;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class ExtendedFields extends MdxBase<ExtendedFields> {
```

Comment:
There's some discussion about these fields in office right now. So it sounds like clients are asking for more fields than we normally support so thats why this object is added? To make account details more extensible/customizable? Is that correct?

Reply:
not more. but HW for example has a lot of account type that we don't support and need certain fields be visible and certain fields be hidden. We came up with this new endpoint to support such requirements. The data wont be persisted anywhere, they are just a pass through.

Reply:
https://developer.mx.com/drafts/mdx/accounts/#accounts-account-details-additional-extended-fields

Greg might have a comment or two, stay tuned.

Reply:
These can be considered as fields that may be some clients specifically needed, which might not be needed for the other clients. This will give the flexibility for future services to send more info about the account to our mobile applications.

Reply:
My issue with the design is that these are generic fields that could be understood/used differently for different clients. This may not matter so much since this is backend code that essentially is passing data through, but ultimately some code somewhere is going to have to understand the data to properly display or store it. So, it seems we're creating a generic specification to push off that need to quantify the result of a request to some future system.

Reply:
Also, have we considered using Protobuf in the backend? They're using it in the frontend for this same code.

Reply:
Mobile App needs to understand this data and display it correctly yes. As part of the discussions with @mattnichols and @JAHansens the long term plan is to have the existing endpoint `Retrieve account details` get trigger each time to user enters that screen but since it's a huge lift off and needs lot of core changes. we all came up with this new temporary approach to unblock a few things for HW.
I think the long term plan will address everything you are concerned about?

Reply:
Then we could generate not just the C++ deserializer for the frontend, but also one for Java.

Reply:
I get that that is a much bigger project.